audformat/core/table.py

```diff
@@ -28,7 +28,499 @@
+            sorted_iter=False,
+            value_type=Column,
+            set_callback=self._set_column,
+        )
+        r"""Table columns"""
+
+        self._df = pd.DataFrame(index=index)
+        self._db = None
+        self._id = None
+
+    def __getitem__(self, column_id: str) -> Column:
+        r"""Return view to a column.
+
+        Args:
+            column_id: column identifier
+
+        """
+        return self.columns[column_id]
+
+    def __eq__(
+            self,
+            other: 'Base',
+    ) -> bool:
+        if self.dump() != other.dump():
+            return False
+        return self.df.equals(other.df)
+
+    def __len__(self) -> int:
+        return len(self.df)
+
+    def __setitem__(self, column_id: str, column: Column) -> Column:
+        r"""Add new column to table.
+
+        Args:
+            column_id: column identifier
+            column: column
+
+        Raises:
+            BadIdError: if a column with a ``scheme_id`` or ``rater_id`` is
+                added that does not exist
+
+        """
+        self.columns[column_id] = column
+        return column
+
+    @property
+    def db(self):
+        r"""Database object.
+
+        Returns:
+            database object or ``None`` if not assigned yet
+
+        """
+        return self._db
+
+    @property
+    def df(self) -> pd.DataFrame:
+        r"""Table data.
+
+        Returns:
+            data
+
+        """
+        if self._df is None:
+            # if database was loaded with 'load_data=False'
+            # we have to load the table data now
+            path = os.path.join(self.db.root, f'{self.db._name}.{self._id}')
+            self.load(path)
+        return self._df
+
+    @property
+    def index(self) -> pd.Index:
+        r"""Table index.
+
+        Returns:
+            index
+
+        """
+        return self.df.index
+
+    @property
+    def media(self) -> typing.Optional[Media]:
+        r"""Media object.
+
+        Returns:
+            media object or ``None`` if not available
+
+        """
+        if self.media_id is not None and self.db is not None:
+            return self.db.media[self.media_id]
+
+    @property
+    def split(self) -> typing.Optional[Split]:
+        r"""Split object.
+
+        Returns:
+            split object or ``None`` if not available
+
+        """
+        if self.split_id is not None and self.db is not None:
+            return self.db.splits[self.split_id]
+
+    def drop_columns(
+            self,
+            column_ids: typing.Union[str, typing.Sequence[str]],
+            *,
+            inplace: bool = False,
+    ) -> 'Table':
+        r"""Drop columns by ID.
+
+        Args:
+            column_ids: column IDs
+            inplace: drop columns in place
+
+        Returns:
+            new ``Table`` if ``inplace=False``, otherwise ``self``
+
+        """
+        if not inplace:
+            return self.copy().drop_columns(column_ids, inplace=True)
```

Comment:
Does this work even though you don't implement `self.copy()`?

Reply:
It works for `audformat.core.table.Table` as you implement `copy()` there, but a developer does not know that she has to implement it, so I think we have to move it to `Base` as well when moving `drop_columns()`

Reply:
Good catch. The problem is that I'm not sure if we can implement it in the base class since we want to create a copy of the derived class. So probably we'll have to make it an abstract method to force the developer to implement it.

Reply:
Turns out it's simply, just use `self.__class__(...)` to construct the object in the base class. Note that I override `copy()` in the child class, but just to get the correct return type in the docs.

Reply:
done